[sqlpp11] Support Conan v2 and add version 0.62

uilianries opened the pull request:
Specify library name and version:  **sqlpp11/0.62**

- Remove 0.5x versions
- Add 0.62 version
- Add support to be built with Conan 2.x
- Update test_package to run with Conan 2.x
- Added add test_v1_package


closes #17992 #17857 #17754

/cc @LvdStokker @CJCombrink 

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

CJCombrink commented:
Also closes #17857 (not sure if it must be added to the main description for it to auto close once done

CJCombrink commented on recipes/sqlpp11/config.yml:
```diff
@@ -1,9 +1,7 @@
 versions:
-  "0.58":
+  "0.62.0":
```
I am guessing this must match [recipes/sqlpp11/all/conandata.yml](https://github.com/conan-io/conan-center-index/pull/17993/files#diff-0b33cd9957ed280dda3ffabc60e391f001ac50cb6dec512711c6fea7d8e112a3) which states `0.62` only

CJCombrink commented:
@uilianries 

> Didn't add test_v1_package

One can probably pull in the v1 test from https://github.com/conan-io/conan-center-index/pull/17754

uilianries commented:
@CJCombrink 

> Also closes https://github.com/conan-io/conan-center-index/issues/17857 (not sure if it must be added to the main description for it to auto close once done)

No problem, I just updated the PR description to close that issue when merging it.

> One can probably pull in the v1 test from https://github.com/conan-io/conan-center-index/pull/17754

Yes, that template we created long time ago, but no longer listed in https://github.com/conan-io/conan-center-index/tree/master/docs/package_templates/header_only/all

> I am guessing this must match [recipes/sqlpp11/all/conandata.yml](https://github.com/conan-io/conan-center-index/pull/17993/files#diff-0b33cd9957ed280dda3ffabc60e391f001ac50cb6dec512711c6fea7d8e112a3) which states 0.62 only

Good catch, fixed!

CJCombrink commented:
@uilianries
Any clues on what went wrong this time?

uilianries commented:
@CJCombrink An internal error occurred in the CI. I just restarted the PR now. Thank you for pinging!

CJCombrink commented:
> @CJCombrink An internal error occurred in the CI. I just restarted the PR now. Thank you for pinging!

@uilianries Seems like it is still not happy

uilianries commented:
@CJCombrink The CI is overloaded as we trying to generate new packages internally. No problem, this PR is marked as `Unexpected Error` and will be built again automatically.

RubenRBS commented on recipes/sqlpp11/all/conanfile.py:
```diff
@@ -12,60 +16,56 @@ class Sqlpp11Conan(ConanFile):
         return {
             "gcc": "5",
             "Visual Studio": "14",
+            "msvc": "14.1",
```
I might be missunderstanding this, but isn't the proper msvc version 190 in this case?

```suggestion
            "msvc": "190",
```

RubenRBS reviewed (commented):
Small question regarding msvc min version, othewise looks great :)

LvdStokker commented:
@uilianries, would you be willing to approve this PR so that I may gain access and speed up the merge of this PR and others in future?

https://github.com/conan-io/conan-center-index/pull/18021#pullrequestreview-1493978631

uilianries commented:
@LvdStokker @CJCombrink The pull request has been merged and the package should be available now in Conan Center. Regards!

CJCombrink commented:
Thanks @uilianries 
Any reason why the linked tickets and PRs did not auto close?

LvdStokker commented:
Thanks a lot @uilianries. Much appreciated. It may still be uploading to Artifactory, I am not certain but I will look out on https://conan.io/center/sqlpp11

uilianries commented:
@CJCombrink Github permission I guess. I just closed them now, thank you for pinging.

prince-chrismc commented:
@uilianries you made a tiny mistake linking issues to close 

You need to write close once for each github issue you want to link :)

uilianries commented:
All right! Thank you for the hint, I'll remember next time!

On Wed, Jun 28, 2023 at 7:25 PM Chris Mc ***@***.***> wrote:

> @uilianries <https://github.com/uilianries> you made a tiny mistake
> linking issues to close
>
> You need to write close once for each github issue you want to link :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/conan-io/conan-center-index/pull/17993#issuecomment-1611810727>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABFFAHJHBGHOMTKHHQ2KEU3XNRSHXANCNFSM6AAAAAAZOL2DTE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Cordialmente,

Uilian Ries
***@***.***>

SpaceIm commented on recipes/sqlpp11/all/conanfile.py:
```diff
@@ -1,8 +1,12 @@
+from conan.tools.build import check_min_cppstd
+from conan.tools.layout import basic_layout
+from conan.tools.files import get, copy
+from conan import Version
```
this import is illegal
